src/main/java/spring/dao/impl/LibraryDaoImpl.java

```diff
@@ -0,0 +1,78 @@
+package spring.dao.impl;
+
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.dao.LibraryDao;
+import spring.entity.Author;
+import spring.entity.Book;
+import spring.entity.User;
+
+import javax.persistence.Query;
+import java.util.List;
+
+@Repository
+public class LibraryDaoImpl implements LibraryDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
+    public List<Book> getBooksRentByUser(User user) {
+        Query queryGetBooks = sessionFactory.createEntityManager().createNativeQuery("select library.books.book_id," +
+                " books.title, books.year, books.price, books.quantity from books" +
+                " inner join rents on rents.book_id = books.book_id " +
+                " where user_id = " + user.getId(), Book.class);
+        return queryGetBooks.getResultList();
+    }
+
+    @Override
+    public void returnBook(User user, Book book) {
+        Integer quantity = book.getQuantity();
+        book.getUsers().remove(user);
+        book.setQuantity(quantity + 1);
+        sessionFactory.getCurrentSession().update(book);
+    }
+
+    @Override
+    public void rentBook(User user, Book book) {
+        Integer quantity = book.getQuantity();
+        if(quantity > 0) {
+            book.getUsers().add(user);
+            book.setQuantity(quantity - 1);
+            sessionFactory.getCurrentSession().update(book);
+        } else {
+            System.out.println("Book isn't available!");
+        }
+    }
+
+    @Override
+    public List<Book> findBooksByAuthor(String name, String surname) {
+        Query queryGetBooks = sessionFactory.createEntityManager().createNativeQuery("select library.books.book_id," +
```

Comment:
too ling line
try using `like` in all query where we find books and authors by name, surname and title